ui/common/src/socket.ts

```diff
@@ -289,57 +283,65 @@ export default class StrongSocket {
     const ws = this.ws;
     if (ws) {
       this.debug('Disconnect');
-      this.autoReconnect = false;
       ws.onerror = ws.onclose = ws.onopen = ws.onmessage = () => {};
       ws.close();
     }
   };
 
-  onError = (e: unknown): void => {
+  private onError = (e: unknown): void => {
+    if (this.heartbeat.wasSuspended) return;
     this.options.debug = true;
     this.debug(`error: ${e} ${JSON.stringify(e)}`); // e not always from lila
   };
 
-  onClose = (e: CloseEvent, url: string): void => {
+  private onClose = (e: CloseEvent): void => {
     site.pubsub.emit('socket.close');
-    if (this.autoReconnect) {
```

Comment:
`this.autoReconnect` is actually redundant with `this.ws` having a value

Reply:
I'm not sure about that. `disconnect` sets `autoReconnect` to false to prevent reconnection. `site.reload` calls `disconnect` to prevent websocket activity while the page is reloading. Sometimes the reload is slow and we don't want the websocket to reconnect while it happens.

Reply:
site.reload calls destroy now. disconnect has been removed from the interface